openfisca_survey_manager/scenarios.py

```diff
@@ -507,10 +514,12 @@ def inflate(self, inflator_by_variable = None, target_by_variable = None):
                     assert column_name in inflator_by_variable, 'column_name is not in inflator_by_variable'
                     log.info('Using inflator {} for {}.  The target is thus {}'.format(
                         inflator_by_variable[column_name],
-                        column_name, inflator_by_variable[column_name] * self.compute_aggregate(variable = column_name))
-                        )
+                        column_name, inflator_by_variable[column_name] * self.compute_aggregate(
+                            variable = column_name, reference = reference, period = period)
+                        ))
                     inflator = inflator_by_variable[column_name]
-
+                
+                assert holder.array is not None
                 holder.array = inflator * holder.array
```

Comment:
I just pushed a fix in Core to repaire the `holder.array` setter, but I'm not sure I understand what this bit of code is supposed to do.
For which period should the inflator be applied ? With `holder.array`, it will only apply to `simulation.period`, not to the `period` argument of the function.

In all case, I'd recommend a more explicit syntax like : 

```py
array = holder.get_array(period)
inflated_array = inflator * array
holder.put_in_cache(inflated_array, period)
```

Reply:
We should definitively use this one. Thanks @fpagnoux